examples/catalyst_contrastive/barlow_twins.py

```diff
@@ -0,0 +1,130 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torch.utils.data import DataLoader
+import torchvision
+from torchvision import transforms
+from torchvision.models.resnet import resnet50
+
+from catalyst import dl
+from catalyst.contrib.nn import BarlowTwinsLoss
+
+
+class CifarPairTransform:
+    def __init__(self, train_transform=True, pair_transform=True):
+        if train_transform is True:
+            self.transform = transforms.Compose(
+                [
+                    transforms.RandomResizedCrop(32),
+                    transforms.RandomHorizontalFlip(p=0.5),
+                    transforms.RandomApply([transforms.ColorJitter(0.4, 0.4, 0.4, 0.1)], p=0.8),
+                    transforms.RandomGrayscale(p=0.2),
+                    transforms.ToTensor(),
+                    transforms.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]),
+                ]
+            )
+        else:
+            self.transform = transforms.Compose(
+                [
+                    transforms.ToTensor(),
+                    transforms.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]),
+                ]
+            )
+        self.pair_transform = pair_transform
+
+    def __call__(self, x):
+        if self.pair_transform is True:
+            y1 = self.transform(x)
+            y2 = self.transform(x)
+            return y1, y2
+        else:
+            return self.transform(x)
+
+
+class Model(nn.Module):
+    def __init__(self, feature_dim=128):
+        super(Model, self).__init__()
+
+        self.f = []
+        for _, module in resnet50().named_children():
+            if not isinstance(module, nn.Linear) and not isinstance(module, nn.MaxPool2d):
+                self.f.append(module)
+
+        # encoder
+        self.f = nn.Sequential(*self.f)
+        # projection head
+        self.g = nn.Sequential(
+            nn.Linear(2048, 512, bias=False),
+            nn.BatchNorm1d(512),
+            nn.ReLU(inplace=True),
+            nn.Linear(512, feature_dim, bias=True),
+        )
+
+    def forward(self, x):
+        x = self.f(x)
+        feature = torch.flatten(x, start_dim=1)
+        out = self.g(feature)
+        return F.normalize(feature, dim=-1), F.normalize(out, dim=-1)
+
+
+class CustomRunner(dl.Runner):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
```

Comment:
and we could also simplify code a bit here 👁️

Reply:
[Done](https://github.com/catalyst-team/catalyst/pull/1261/commits/497fc398fa989217a3c4b1752ef76d78da24bcde)